datamodel/core.py

```diff
@@ -343,6 +348,15 @@ def _build_datamodel(cls, registry):
 
         return DatamodelClass
 
+    @classmethod
+    def _get_env_from_registry(cls, registry):
```

Comment:
@kevinkhao Your proposal doesn't work. Indeed the env from datamodels will be into an other transaction and with the wrong user.....